include/mouse.h

```diff
@@ -22,39 +22,79 @@
 
 #include "dosbox.h"
 
+#include <regex>
+#include <string>
+#include <vector>
+
+// ***************************************************************************
+// Initialization, configuration
+// ***************************************************************************
+
+void MOUSE_Init(Section * );
+void MOUSE_AddConfigSection(const config_ptr_t &);
+
+// ***************************************************************************
+// Data types
+// ***************************************************************************
+
+enum class MouseInterfaceId : uint8_t {
+    DOS,  // emulated DOS mouse driver
+    PS2,  // PS/2 mouse (this includes VMware mouse protocol)
+    COM1, // serial mouse
+    COM2,
+    COM3,
+    COM4,
+
+    First = DOS,
+    Last  = COM4,
+    None  = UINT8_MAX
+};
+
+constexpr uint8_t num_mouse_interfaces_com = 4;
```

Comment:
Specifying four COM ports here feels like an architectural island managing its own state of the world, when `include/serialport.h` is used by the serial port implementation, which has stuff like:

```c++
#define SERIAL_MAX_PORTS  4
const uint8_t serial_defaultirq[] = {4, 3, 4, 3};
const uint16_t serial_baseaddr[] = {0x3f8, 0x2f8, 0x3e8, 0x2e8};
const char *const serial_comname[] = {"COM1", "COM2", "COM3", "COM4"};
```

I faced a similar issue with the three LPT port, when refactoring the DACs, so I placed the generic LPT port info in `include/lpt.h`, which can be used by anything involving the printer ports.

Maybe there's a similar solution here, where `serial.h` and `mouse.h` both get some very basic facts about the COM ports from `comports.h`.

And then in mouse, you could have a strong programmatic link to those `ComPorts` enum class(es).

``` c++
enum class MouseInterfaceId : uint8_t {
    DOS,  // emulated DOS mouse driver
    PS2,  // PS/2 mouse (this includes VMware mouse protocol)
    COM1 = ComPorts::COM1, // serial mouse
    COM2 = ComPorts::COM2,
    COM3 = ComPorts::COM3,
    COM4 = ComPorts::COM4,
```

(Or something to that effect)

Reply:
Got rid of `num_mouse_interfaces_com`, reused `SERIAL_MAX_PORTS`. This is actually the only COM port related value we need to know here. Reusing port identifier from other enum class wouldn't be practical - we want to have continuous numbering (so that we can use arrays), we can't have overlapping ID's, and our number of mouse interfaces is likely to increase in the future - as InPort/Bus mouse is still not implemented.